packages/liveblocks-react-lexical/src/comments/threads-panel.tsx

```diff
@@ -0,0 +1,429 @@
+    for (const { thread, rect } of ascending) {
+      let top = rect.top - container.getBoundingClientRect().top;
+
+      for (const [id, position] of newPositions) {
+        // Retrieve the element associated with the thread
+        const el = elements.get(id);
+        if (el === undefined) continue;
+
+        if (
+          top >= position &&
+          top <= position + el.getBoundingClientRect().height
+        ) {
+          top = position + el.getBoundingClientRect().height;
+        }
+      }
+
+      newPositions.set(thread.id, top);
+    }
+
+    for (const { thread, rect } of descending.reverse()) {
+      // Retrieve the element associated with the current thread
+      const el = elements.get(thread.id);
+      if (el === undefined) continue;
+
+      let top = rect.top - container.getBoundingClientRect().top;
+      for (const [, position] of newPositions) {
+        if (top >= position - el.getBoundingClientRect().height) {
+          top = position - el.getBoundingClientRect().height;
+        }
+      }
+
+      newPositions.set(thread.id, top);
+    }
+
+    setPositions(newPositions);
+  }, [editor, threads, nodes, activeThreads, elements]);
+
+  useLayoutEffect(() => {
+    handlePositionThreads();
+  }, [handlePositionThreads]);
+
+  useEffect(() => {
+    return editor.registerUpdateListener(() => {
+      handlePositionThreads();
+    });
+  }, [editor, handlePositionThreads]);
+
+  useEffect(() => {
+    const observer = new ResizeObserver(handlePositionThreads);
+    for (const element of elements.values()) {
+      observer.observe(element);
+    }
+
+    return () => observer.disconnect();
+  }, [elements, handlePositionThreads]);
+
+  const root = useRootElement();
+
+  useEffect(() => {
+    if (root === null) return;
+    const observer = new ResizeObserver(handlePositionThreads);
+
+    observer.observe(root);
+    return () => observer.disconnect();
+  }, [root, handlePositionThreads]);
+
+  if (orderedThreads.length === 0) return;
+
+  return (
+    <div
+      {...divProps}
+      className={classNames(className, "lb-root lb-lexical-thread-panel")}
+      ref={containerRef}
+      style={{
+        position: "relative",
+        ...style,
+      }}
+    >
+      {orderedThreads.map(({ thread, rect }) => {
+        let top = rect.top;
+
+        if (positions.has(thread.id)) {
+          top = positions.get(thread.id)!;
+        }
+
+        const isActive = activeThreads.includes(thread.id);
+
+        return (
+          <ThreadWrapper
+            key={thread.id}
+            Thread={Thread}
+            thread={thread}
+            onItemAdd={onItemAdd}
+            onItemRemove={onItemRemove}
+            style={{
+              position: "absolute",
+              transform: `translateX(${isActive ? -12 : 0}px) translateY(${top}px)`,
+              left: 0,
+              width: "100%",
+              paddingBottom: gap,
+              transition: "transform 0.2s",
```

Comment:
@marcbouchenoire This will likely need to be removed but if we choose to control this animations via JavaScript in future, should we make this default for now so that users aren't surprised when they see threads being animated? 🤔

Reply:
Regardless of this I think it belongs in the external styles if we want to keep it by default